File: security/src/main/java/net/yan100/compose/security/properties/JwtProperties.java
/*
 *  Copyright (c) 2020-2024 TrueNine. All rights reserved.
 * 
 * The following source code is owned, developed and copyrighted by TrueNine
 * (truenine304520@gmail.com) and represents a substantial investment of time, effort,
 * and resources. This software and its components are not to be used, reproduced,
 * distributed, or sublicensed in any form without the express written consent of
 * the copyright owner, except as permitted by law.
 * Any unauthorized use, distribution, or modification of this source code,
 * or any portion thereof, may result in severe civil and criminal penalties,
 * and will be prosecuted to the maximum extent possible under the law.
 * For inquiries regarding usage or redistribution, please contact:
 *     TrueNine
 *     email: <truenine304520@gmail.com>
 *     website: <github.com/TrueNine>
*/
package net.yan100.compose.security.properties;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

@Data
@ConfigurationProperties(prefix = "compose.security.jwt")
public class JwtProperties {
    String publicKeyClassPath = "security/pub.key";
    String privateKeyClassPath = "security/pri.key";
    String encryptDataKeyName = "edt";
    String issuer = "T-SERVER";
    Long expiredDuration = (long) (2 * 60 * 60 * 60 * 1000);
}

File: security/spring/src/main/kotlin/io/github/truenine/composeserver/security/properties/JwtProperties.kt
package io.github.truenine.composeserver.security.properties

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConfigurationPropertiesBinding

private const val PREFIX = "compose.security"

@ConfigurationPropertiesBinding
@ConfigurationProperties(prefix = "$PREFIX.jwt")
data class JwtProperties(
  var publicKeyClassPath: String = "security/pub.key",
  var privateKeyClassPath: String = "security/pri.key",
  var encryptDataKeyName: String = "edt",
  var issuer: String = "T-SERVER",
  var expiredDuration: Long = (2 * 60 * 60 * 60 * 1000).toLong(),
)
